Add doc about collecting trace with assembly loading information

elinor-fung opened the pull request:
## Summary

Add documentation about collecting a trace with detailed assembly loading information

elinor-fung committed:
Add doc about collecting trace with assembly loader events

vitek-karas reviewed (approved):
Great writeup!

elinor-fung committed:
Apply suggestions from code review

Co-authored-by: Vitek Karas <vitek.karas@microsoft.com>

elinor-fung committed:
Add names for event data with enum values

IEvangelist reviewed (changes requested):
We need to add these to the TOC. See https://github.com/dotnet/docs/blob/master/docs/fundamentals/toc.yml

elinor-fung committed:
Update table of contents

adegeo commented on docs/core/dependency-loading/collect-details.md:
```diff
@@ -0,0 +1,164 @@
+
+Open the collected trace in [PerfView](https://github.com/microsoft/perfview) and open the Events view. Filter the events list to `Microsoft-Windows-DotNETRuntime/AssemblyLoader` events.
+
+![PerfView assembly loader filter image](media/collect-details/assembly-loader-filter.png)
```
Please review all of your images. You should use the new `:::image` syntax. This automatically adds a grey border around the image as required. Also, make sure your images are <= 1200 pixels. If you can't fit the content in that size, try using a 100% zoom level on your desktop.
```suggestion
:::image type="content" source="media/collect-details/assembly-loader-filter.png" alt-text="PerfView assembly loader filter image.":::
```

adegeo commented on docs/core/dependency-loading/collect-details.md:
```diff
@@ -0,0 +1,164 @@
+
+| Event Name                                                | AssemblyName                                    | HandlerName                    | ResultAssemblyPath |
+|-----------------------------------------------------------|-------------------------------------------------|--------------------------------|--------------------|
+| AssemblyLoader/AssemblyLoadContextResolvingHandlerInvoked | MyLibrary, Culture=neutral, PublicKeyToken=null | OnAssemblyLoadContextResolving | C:\samples\core\extensions\AssemblyLoading\AssemblyLoading\bin\Debug\net5.0\MyLibrary.dll |
```
There are a few references to file/folder paths in the articles. File paths should be enclosed in _italics_

adegeo reviewed (changes requested):
Thank you for the articles! This content will be really helpful to our users. I have a few things that need to be addressed before we can move forward.

elinor-fung committed:
Smaller images

elinor-fung committed:
PR feedback

elinor-fung commented on docs/core/dependency-loading/assembly-loader-events.md:
```diff
@@ -0,0 +1,96 @@
+---
+title: Assembly Loader Events
```
Yes, it is related to diagnostics. The events would go through EventPipe, ETW, and LTTng. I had added this page here with the intent of removing it once we had a place with all the runtime events. Since #20817 is already out for adding that, I'll remove this page and hold off on this change so that we can just link to the content from #20817 once it is in.

elinor-fung committed:
Link to docs for runtime provider events

elinor-fung commented:
I fixed the images and updated this change to point to the new docs about runtime events that were just merged. This should be ready for another review now.

@adegeo @IEvangelist Could you take another look?

IEvangelist reviewed (approved):
LGTM - I'll let @adegeo approve it from here and merge. Thanks

adegeo reviewed (approved):
Sorry for the delay!
